lldb/unittests/Symbol/TestSwiftASTContext.cpp

```diff
@@ -199,3 +199,32 @@ TEST_F(TestSwiftASTContext, IsNonTriviallyManagedReferenceType) {
                                                                    nullptr));
 #endif
 }
+
+namespace {
+  const std::vector<std::string> duplicated_flags = {
+    "-DMACRO1", "-D", "MACRO1", "-UMACRO2", "-U", "MACRO2",
+    "-I/path1", "-I", "/path1", "-F/path2", "-F", "/path2",
+    "-fmodule-map-file=/path3", "-fmodule-map-file=/path3",
+    "-F/path2", "-F", "/path2", "-I/path1", "-I", "/path1",
+    "-UMACRO2", "-U", "MACRO2", "-DMACRO1", "-D", "MACRO1",
+  };
+  const std::vector<std::string> unqued_flags = {
+    "-DMACRO1", "-UMACRO2", "-I/path1", "-F/path2", "-fmodule-map-file=/path3"
+  };
+} // namespace
+
+TEST(ClangArgs, UniquingCollisionWithExistingFlags) {
+  const std::vector<std::string> source = duplicated_flags;
+  std::vector<std::string> dest = unqued_flags;
+  lldb_private::AddExtraClangArgs(source, dest);
+
+  EXPECT_EQ(dest, unqued_flags);
+}
+
+TEST(ClangArgs, UniquingCollisionWithAddedFlags) {
+  const std::vector<std::string> source = duplicated_flags;
+  std::vector<std::string> dest;
+  lldb_private::AddExtraClangArgs(source, dest);
+
+  EXPECT_EQ(dest, unqued_flags);
```

Comment:
This is great!

Reply:
Too bad that there is a typo on this exact line :)